target_chains/aptos/contracts/sources/pyth.move

```diff
@@ -1430,4 +1527,158 @@ module pyth::pyth_test {
+
+        let update_data = TEST_VAAS;
+        let min_publish_time: u64 = 1663074345;
+        let max_publish_time: u64 = 1663680750;
+
+        let initial_pyth_balance = coin::balance<AptosCoin>(@pyth);
+        let fee_value = coin::value(&coins);
+
+        let test_price_feeds: vector<pyth::price_feed::PriceFeed> = pyth::parse_price_feed_updates(update_data, min_publish_time, max_publish_time, coins);
+        let mock_price_infos: vector<pyth::price_info::PriceInfo> = get_mock_price_infos();
+
+        // update fee transfer validation
+        let final_pyth_balance = coin::balance<AptosCoin>(@pyth);
+        assert!(final_pyth_balance == initial_pyth_balance + fee_value, 1);
+        
+        assert!(vector::length(&test_price_feeds) > 0, 1);
+        let i: u64 = 0;
+        while (i < vector::length(&mock_price_infos)) {
+            let mock_price_info: &pyth::price_info::PriceInfo = vector::borrow(&mock_price_infos, i);
+            let test_price_feed: &pyth::price_feed::PriceFeed = vector::borrow(&test_price_feeds, i);
+            let mock_price_feed: &pyth::price_feed::PriceFeed = price_info::get_price_feed(mock_price_info);
+
+            let test_price_id: &price_identifier::PriceIdentifier = price_feed::get_price_identifier(test_price_feed);
+            let mock_price_id: &price_identifier::PriceIdentifier = price_feed::get_price_identifier(mock_price_feed);
+            let test_price: price::Price = price_feed::get_price(test_price_feed);
+            let mock_price: price::Price = price_feed::get_price(mock_price_feed);
+            let test_ema_price: price::Price = price_feed::get_ema_price(test_price_feed);
+            let mock_ema_price: price::Price = price_feed::get_ema_price(mock_price_feed);
+
+            assert!(test_price_id == mock_price_id, 1);  
+            assert!(test_price == mock_price, 1);    
+            assert!(test_ema_price == mock_ema_price, 1); 
+                  
+            i = i + 1;
+        };
+        
+        cleanup_test(burn_capability, mint_capability);
+    }
+
+
+    #[test(aptos_framework = @aptos_framework)]
+    #[expected_failure(abort_code = 65542, location = pyth::pyth)]
+    fun test_parse_price_feed_updates_insufficient_fee(aptos_framework: &signer) {
+        let update_fee = 50;
+        let initial_balance = 20;
+        let (burn_capability, mint_capability, coins) = setup_test(aptos_framework, 500, 1,
+            x"5d1f252d5de865279b00c84bce362774c2804294ed53299bc4a0389a5defef92",
+            data_sources_for_test_vaa(),
+            update_fee,
+            initial_balance);
+
+        let update_data = TEST_VAAS;
+        let min_publish_time: u64 = 1663074345;
+        let max_publish_time: u64 = 1663680750;
+
+        let test_price_feeds: vector<pyth::price_feed::PriceFeed> = pyth::parse_price_feed_updates(update_data, min_publish_time, max_publish_time, coins);        
+        assert!(vector::length(&test_price_feeds) == 0, 1);
```

Comment:
I don't think the test reaches this line since the previous line will fail. We can probably remove it for clarity. the same goes for the next few testcases where we expect failure